src/System/Components/RouterLink.tsx

```diff
@@ -84,11 +88,31 @@ export const RouterLink: React.FC<React.PropsWithChildren<React.PropsWithChildre
       }
     }
 
+    // Inject a query param to indicate that the link was successfully pre-fetched.
+    const updatedHrefWithPrefetchQueryParam = (
+      to: string,
+      isPrefetched: boolean
+    ): string => {
+      const [baseUrl, queryString] = to.split("?")
+      const queryParams = new URLSearchParams(queryString || "")
+      if (isPrefetched) {
+        queryParams.set("is_prefetched", "true") // intentional underscore casing
```

Comment:
We have some middleware that rewrites query params from camel case to underscore (?) - just to avoid anything weirdness I rolled with underscore casing on the query param to start.

Reply:
Whats up with the underscore middleware? Is this definitely necessary?

Reply:
If underscore is required -- we should definitely fix this -- then we should at least 1:1 the param name to the variable: `is_prefetching` (but really we should fix this. Devs should certainly be able to write any kind of query param they want, without conditions)

Reply:
I couldn't figure out where that was happening! You can see the behavior by going to a page like https://staging.artsy.net/artist/andy-warhol?isPrefetched=true , and watch the address bar - it updates.

So I figured to avoid anything annoying there and just use underscore for now.